### PR TITLE
Clarify Style Guide Regarding Third-Party Content

### LINKED ISSUE #15576: Clarify style guide: what kinds of third party content do K8s docs allow?
## Improve the K8s style guide

The style guide needs to specify that K8s docs decline to host specific kinds of content:

1. Provider- or project-specific content from projects outside the kubernetes or kubernetes-sigs GitHub organizations

2. Identical content hosted in multiple locations ("dual-sourced" content)

For example, consider [Kubernetes in Docker](https://github.com/kubernetes-sigs/kind) (KinD):

Allowed | Action | Reason
---|---|---
✔️ | Linking to KinD docs from K8s docs | The KinD project resides in `github.com/kubernetes-sigs`.
❌ | Hosting KinD docs in k/website | Dual-sourced content requires duplicated effort from maintainers and tends to go stale/rotten more quickly.

## Why does this matter?

It's necessary to clarify which kinds of content we do and don't accept in K8s docs.

/sig docs
/priority important-soon

### PR DESCRIPTION
**Pull request by aimeeu**
Clarify style guide regarding what kinds of third-party content are allowed, with examples.

Fixes #15576 
/assign @zacharysarah 
/assign @sftim

### CODE REVIEW

File: content/en/docs/contribute/style/style-guide.md [hunk 1]
```diff
@@ -38,6 +38,30 @@ The English-language documentation uses U.S. English spelling and grammar.
 
 {{< comment >}}[If you're localizing this page, you can omit the point about US English.]{{< /comment >}}
 
+## Documentation content standards
+
+Kubernetes documentation does not host certain types of content:
```
Comment (zacharysarah) [inline content/en/docs/contribute/style/style-guide.md]: Rephrase this to use fewer bullets and [simpler, more direct language](https://kubernetes.io/docs/contribute/style/style-guide/#use-simple-and-direct-language).

Also, we can clarify that we host content from *current* CNCF projects.
```
Kubernetes documentation only hosts content from projects within the `kubernetes` or `kubernetes-sigs` GitHub organizations, or from current CNCF projects.

Kubernetes documentation does not host content identical content sourced from multiple locations ("dual-sourced" content). 
```

File: content/en/docs/contribute/style/style-guide.md [hunk 2]
```diff
@@ -38,6 +38,30 @@ The English-language documentation uses U.S. English spelling and grammar.
...
+* Identical content hosted in multiple locations ("dual-sourced" content)
+
+For example:
+
+| Allowed |                           Action                          |                                                    Reason                                                   |
+|:-------:|:----------------------------------------------------------|:------------------------------------------------------------------------------------------------------------|
+|    Y    | Linking to Kubernetes in Docker (KinD) docs from k8s docs | The KinD project resides in github.com/kubernetes-sigs.                                                     |
+|    N    | Hosting KinD docs in k8s website                          | Dual-sourced content requires duplicated effort from maintainers and tends to go stale/rotten more quickly. |
+|    Y    | Linking to Prometheus docs from k8s docs                  | Prometheus is a CNCF project.                                                                                |
+|    N    | Linking to rkt docs from k8s docs                         | rkt is an (almost) archived CNCF project.                                                                    |
```
Comment (zacharysarah) [inline content/en/docs/contribute/style/style-guide.md]: We can leave out "(almost)" here, to avoid [statements about the future](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-statements-about-the-future).

Comment (zacharysarah) [inline content/en/docs/contribute/style/style-guide.md]: I think the explanation here needs to be clearer for newer contributors. So something like:
> rkt is an archived CNCF project and no longer current

File: content/en/docs/contribute/style/style-guide.md [hunk 3]
```diff
@@ -38,6 +38,30 @@ The English-language documentation uses U.S. English spelling and grammar.
...
+|    N    | Tutorial on using a specific product to monitor Kubernetes | Third-party product-specific content not allowed; this type of content belongs in the third-party product's documentation. |
+|    Y    | Linking to vendor-specific product docs when listing third-party products such as Ingress controllers, load-balancing products, monitoring tools, and device plugins | Avoids dual-sourced content
```
Comment (zacharysarah) [inline content/en/docs/contribute/style/style-guide.md]: ...but only if they're inside the kubernetes or kubernetes-sigs orgs, or inside the CNCF. Consider omitting this example because it may be more confusing than helpful.

File: content/en/docs/contribute/style/style-guide.md [hunk 4]
```diff
@@ -38,6 +38,30 @@ The English-language documentation uses U.S. English spelling and grammar.
...
+|    Y    | Linking to an online training course                      |  Avoids dual-sourced content                                                                                  |
```
Comment (zacharysarah) [inline content/en/docs/contribute/style/style-guide.md]: There's maybe [one page](https://github.com/kubernetes/website/blob/master/content/en/docs/setup/_index.md) where we accept this kind of content. Consequently, please omit this example.

File: content/en/docs/contribute/style/style-guide.md [hunk 5]
```diff
@@ -38,6 +38,30 @@ The English-language documentation uses U.S. English spelling and grammar.
 
 {{< comment >}}[If you're localizing this page, you can omit the point about US English.]{{< /comment >}}
 
+## Documentation content standards
```
Comment (zacharysarah) [inline content/en/docs/contribute/style/style-guide.md]: Change this to something clearer:
> ## What Kubernetes documentation does and doesn't contain

File: content/en/docs/contribute/style/style-guide.md [hunk 6]
```diff
@@ -38,6 +38,30 @@ The English-language documentation uses U.S. English spelling and grammar.
...
+|:-------:|:----------------------------------------------------------|:------------------------------------------------------------------------------------------------------------|
+|    Y    | Linking to Kubernetes in Docker (KinD) docs from k8s docs | The KinD project resides in github.com/kubernetes-sigs.                                                     |
+|    N    | Hosting KinD docs in k8s website                          | Dual-sourced content requires duplicated effort from maintainers and tends to go stale/rotten more quickly. |
+|    Y    | Linking to Prometheus docs from k8s docs                  | Prometheus is a CNCF project.                                                                                |
+|    N    | Linking to rkt docs from k8s docs                         | rkt is an (almost) archived CNCF project.                                                                    |
+|    N    | Tutorial on using a specific product to monitor Kubernetes | Third-party product-specific content not allowed; this type of content belongs in the third-party product's documentation. |
```
Comment (zacharysarah) [inline content/en/docs/contribute/style/style-guide.md]: - Make sure example format stays consistent.
- Use complete sentences.
- Use simple and natural language.
- Make sure examples are clear.

> No | Hosting a tutorial for a specific product, whether inside or outside the CNCF | Dual-sourced content isn't allowed.

**Comment by zacharysarah** [comment]
Preview link: https://deploy-preview-15774--kubernetes-io-master-staging.netlify.com/docs/contribute/style/style-guide/#documentation-content-standards

File: content/en/docs/contribute/style/style-guide.md [hunk 7]
```diff
@@ -38,6 +38,30 @@ The English-language documentation uses U.S. English spelling and grammar.
...
+* Vendor, provider, or project-specific content from projects that are not:
+
+    * Inside the kubernetes or kubernetes-sigs GitHub organizations
+    * Inside the CNCF
+
+* Identical content hosted in multiple locations ("dual-sourced" content)
+
+For example:
+
+| Allowed |                           Action                          |                                                    Reason                                                   |
```
[the remark this answers is not in the document]
Comment (sftim) [reply content/en/docs/contribute/style/style-guide.md]: My gut feeling is that this would be easier to read in the order Action, Allowed, Reason.
@aimeeu to be clear I'm also fine with the current ordering, this feedback wouldn't block me giving an LGTM.

File: content/en/docs/contribute/style/style-guide.md (hunk 2 above)
Comment (aimeeu) [reply content/en/docs/contribute/style/style-guide.md]: @zacharysarah  how do writers and developers know rkt is going to be archived? I checked rkt's [roadmap](https://github.com/rkt/rkt/blob/master/ROADMAP.md) but I didn't see anyting about an archive date, and rkt is listed an an incubating project on the CNCF project [page](https://www.cncf.io/projects/). Does the CNCF maintain a list of soon-to-be-archived projects? I'd like to add verbiage on where to check for soon-to-be-archived CNCF projects.

File: content/en/docs/contribute/style/style-guide.md (hunk 3 above)
Comment (aimeeu) [reply content/en/docs/contribute/style/style-guide.md]: @zacharysarah I misunderstood the extent to which vendor-specific content is not allowed until I read the Slack thread from Friday, 9 August. I'll omit the example.

File: content/en/docs/contribute/style/style-guide.md (hunk 4 above)
Comment (aimeeu) [reply content/en/docs/contribute/style/style-guide.md]: [Tutorials/OnlineTrainingCources/Overview...](https://kubernetes.io/docs/tutorials/online-training/overview/) lists close to 30 online training courses. So I think it would be better to change "Linking to an online training course" to N with an appropriate  reason.

File: content/en/docs/contribute/style/style-guide.md [hunk 8]
```diff
@@ -38,6 +38,30 @@ The English-language documentation uses U.S. English spelling and grammar.
...
+|:-------:|:----------------------------------------------------------|:------------------------------------------------------------------------------------------------------------|
+|    Y    | Linking to Kubernetes in Docker (KinD) docs from k8s docs | The KinD project resides in github.com/kubernetes-sigs.                                                     |
```
[the remark this answers is not in the document]
Comment (aimeeu) [reply content/en/docs/contribute/style/style-guide.md]: done

**Comment by aimeeu** [comment]
@zacharysarah  question on how tables are coded in the style-guide.md file.  I just noticed (ka-thunk) that I used a markdown table whereas the rest of the tables on the page are HTML tables. The MD table header is rendered with a dark background whereas the HTML table header has a white background ([link](https://deploy-preview-15774--kubernetes-io-master-staging.netlify.com/docs/contribute/style/style-guide/)). The style guide doesn't actually specify which table type is preferred. So should I change my markdown table to HTML or change the HTML tables on the page to markdown tables?

File: content/en/docs/contribute/style/style-guide.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (aimeeu) [reply content/en/docs/contribute/style/style-guide.md]: expanded

File: content/en/docs/contribute/style/style-guide.md [hunk 9]
```diff
@@ -38,6 +38,30 @@ The English-language documentation uses U.S. English spelling and grammar.
...
+|    N    | Hosting KinD docs in k8s website                          | Dual-sourced content requires duplicated effort from maintainers and tends to go stale/rotten more quickly. |
```
[the remark this answers is not in the document]
Comment (aimeeu) [reply content/en/docs/contribute/style/style-guide.md]: done

File: content/en/docs/contribute/style/style-guide.md (hunk 2 above)
Comment (aimeeu) [reply content/en/docs/contribute/style/style-guide.md]: done

File: content/en/docs/contribute/style/style-guide.md (hunk 1 above)
Comment (aimeeu) [reply content/en/docs/contribute/style/style-guide.md]: updated

File: content/en/docs/contribute/style/style-guide.md (hunk 2 above)
Comment (aimeeu) [reply content/en/docs/contribute/style/style-guide.md]: changed

File: content/en/docs/contribute/style/style-guide.md (hunk 3 above)
Comment (aimeeu) [reply content/en/docs/contribute/style/style-guide.md]: removed

File: content/en/docs/contribute/style/style-guide.md (hunk 5 above)
Comment (aimeeu) [reply content/en/docs/contribute/style/style-guide.md]: changed

File: content/en/docs/contribute/style/style-guide.md (hunk 6 above)
Comment (aimeeu) [reply content/en/docs/contribute/style/style-guide.md]: changed

File: content/en/docs/contribute/style/style-guide.md [hunk 10]
```diff
@@ -38,29 +38,25 @@ The English-language documentation uses U.S. English spelling and grammar.
 
 {{< comment >}}[If you're localizing this page, you can omit the point about US English.]{{< /comment >}}
 
-## Documentation content standards
+## What Kubernetes documentation does and doesn't contain
```
Comment (kbhawkey) [inline content/en/docs/contribute/style/style-guide.md]: hello @aimeeu . Thanks for working on the style guide.
nit: Could the title be shortened to something like: `Guidelines for contributing content`

File: content/en/docs/contribute/style/style-guide.md [hunk 11]
```diff
@@ -38,29 +38,25 @@ The English-language documentation uses U.S. English spelling and grammar.
...
 
-Kubernetes documentation does not host certain types of content:
+Kubernetes documentation only hosts content from projects within the
+ `kubernetes` or `kubernetes-sigs` GitHub organizations or from current CNCF
+  projects.
```
Comment (kbhawkey) [inline content/en/docs/contribute/style/style-guide.md]: nit: Could provide a link to the list of current CNCF projects.

File: content/en/docs/contribute/style/style-guide.md [hunk 12]
```diff
@@ -38,29 +38,25 @@ The English-language documentation uses U.S. English spelling and grammar.
...
 
-* Vendor, provider, or project-specific content from projects that are not:
-
-    * Inside the kubernetes or kubernetes-sigs GitHub organizations
-    * Inside the CNCF
-
-* Identical content hosted in multiple locations ("dual-sourced" content)
+Kubernetes documentation does not host identical content sourced from
+ multiple locations ("dual-sourced" content).
```
Comment (kbhawkey) [inline content/en/docs/contribute/style/style-guide.md]: nit: If the content is not identical is the content still considered `dual-sourced`?

File: content/en/docs/contribute/style/style-guide.md [hunk 13]
```diff
@@ -38,29 +38,25 @@ The English-language documentation uses U.S. English spelling and grammar.
...
-|    Y    | Linking to an online training course                      |  Avoids dual-sourced content                                                                                  |
-
+| Allowed |                           Action                                              |                                                    Reason                                                   |
+|:-------:|:------------------------------------------------------------------------------|:------------------------------------------------------------------------------------------------------------|
+|    Yes  | Linking to Kubernetes in Docker (KinD) docs from k8s docs                     | The KinD project resides in `kubernetes-sigs` GitHub organization.                                                     |
+|    No   | Hosting KinD docs in k8s website                                              | Dual-sourced content requires duplicated effort from project maintainers and tends to go stale/rotten more quickly. |
```
Comment (kbhawkey) [inline content/en/docs/contribute/style/style-guide.md]: nit: The `Action` column could be the first column.
nit: The table could use an intro paragraph explaining the rules for contributing acceptable content. The table then provides examples of this content.
nit: Hosting [KinD](...) docs `from the` Kubernetes website; link to KinD?
nit: Line 59, Is linking to a CNCF or Linux Foundation online training course permitted?
nit: Kubernetes docs vs Kubernetes Docs?

Comment (aimeeu) [reply content/en/docs/contribute/style/style-guide.md]: RE inking to a CNCF or Linux Foundation online training course permitted - I would say no, because those links can and do go stale. Courses are removed from current offerings, updated with a name change, and the like.

File: content/en/docs/contribute/style/style-guide.md (hunk 11 above)
Comment (aimeeu) [reply content/en/docs/contribute/style/style-guide.md]: @kbhawkey  - yes that was on my list of things to add.

File: content/en/docs/contribute/style/style-guide.md (hunk 13 above)
Comment (aimeeu) [reply content/en/docs/contribute/style/style-guide.md]: @kbhawkey excellent eye! Thanks for the feedback!

File: content/en/docs/contribute/style/style-guide.md (hunk 12 above)
Comment (aimeeu) [reply content/en/docs/contribute/style/style-guide.md]: @kbhawkey Based on my conversations with @zacharysarah, I'd say yes. I changed "identical" to "duplicate"  - not much difference I know.

File: content/en/docs/contribute/style/style-guide.md (hunk 11 above)
Comment (aimeeu) [reply content/en/docs/contribute/style/style-guide.md]: @kbhawkey added links to cncf projects and sandbox projects as well as how to find out if a cncf project has been archived

File: content/en/docs/contribute/style/style-guide.md (hunk 10 above)
Comment (aimeeu) [reply content/en/docs/contribute/style/style-guide.md]: Updated - I like your suggestion

**Comment by sftim** [comment]
> should I change my markdown table to HTML or change the HTML tables on the page to markdown tables to all tables are visually the same style?

It sounds like there's a low-priority issue to be logged, about making sure that readers see one consistent style for tables in the documentation (unless there's a reason for a difference).
Does that sound right?

**Comment by aimeeu** [comment]
@sftim - Yes. I will create an issue once this PR has merged.

> > should I change my markdown table to HTML or change the HTML tables on the page to markdown tables to all tables are visually the same style?
> 
> It sounds like there's a low-priority issue to be logged, about making sure that readers see one consistent style for tables in the documentation (unless there's a reason for a difference).
> Does that sound right?

**Comment by aimeeu** [comment]
replaced by #15892 because I had severe git issues trying to squash commits
